cli/webhook.py

```diff
@@ -0,0 +1,125 @@
+            if 'slack_deploy_botid' in roger_env.keys():
+                self.botid = roger_env['slack_deploy_botid']
+        except (Exception) as e:
+            print("Warning: slackweb basic initialization failed (error: %s).\
+            Not using slack." % e)
+            return
+        try:
+            self.sc = SlackClient(self.token)
+            self.client = slackweb.Slack(url=self.webhookURL)
+            self.disabled = False
+        except (Exception) as e:
+            print("Warning: slackweb basic initialization failed (error: %s).\
+            Not using slack." % e)
+            return  # disabled flag remains False
+
+    def custom_api_call(self, text, channel):
+        self.webhookSetting()
+        if len(channel) == 0:
+            channel = self.defChannel
+        var = self.sc.api_call("channels.list")
+        length = len(var['channels'])
+        for iterator in range(0, length):
+            # getting rid of # for comparison
+            if channel[1:] == var['channels'][iterator]['name']:
+                if self.botid in var['channels'][iterator]['members']:
+                    if not self.disabled:
+                        self.client.notify(channel=channel, username=self.username,
+                                           icon_emoji=self.emoji, text=text)
+
+    def invoke_webhook(self, appdata, hook_input_metric):
+        self.webhookSetting()
+        try:
+            action = app_name = envr = user = ''
+            temp = hook_input_metric.split(',')
+            for var in temp:
+                varAnother = var.split('=')[0]
+                if varAnother == 'event':
+                    action = var.split('=')[1]
+                    continue
+
+                if varAnother == 'app_name':
+                    app_name = var.split('=')[1]
+                    continue
+
+                if varAnother == 'env':
+                    envr = var.split('=')[1]
+                    continue
+
+                if varAnother == 'user':
+                    user = var.split('=')[1]
+                    continue
+            if len(action) == 0 or len(app_name) == 0 or len(envr) == 0 or len(user) == 0:
+                raise ValueError
+
+        except (Exception, KeyError, ValueError, IndexError) as e:
+            print("The following error occurred: %s" %
+                  e)
+            raise
+        try:
+            if 'notifications' in appdata:
+                channelsSet = Set(appdata['notifications']['channels'])
+                envSet = Set(appdata['notifications']['envs'])
+                commandsSet = Set(appdata['notifications']['commands'])
+                # to handle no tag at all
+                if (len(channelsSet) == 0 or len(envSet) == 0 or len(commandsSet) == 0):
+                    print("Notificaton tag missing. Not posting message to slack!")
+                    return
+            else:
+                print("Notificaton tag missing. Not posting message to slack!")
```

Comment:
If notification tag is missing, assume none is needed. So, need to print this message.